Services/source/housekeeping/housekeeping_to_beacon.c

```diff
@@ -0,0 +1,162 @@
+  //Populate the beacon packet by fetching relevant data from housekeeping_services.c
+  //get the unix time from RTC, and convert it to a struct using RTCMK_GetUnix
+  //RTCMK_GetUnix(&(beacon_packet.time));
+  memcpy(&(beacon_packet.time), &(all_hk_data->hk_timeorder.UNIXtimestamp), sizeof(beacon_packet.time));
+ 
+  /*-------EPS-------*/
+  memcpy(&(beacon_packet.eps_mode), &(all_hk_data->EPS_hk.battMode), sizeof(beacon_packet.eps_mode));
+  memcpy(&(beacon_packet.battery_voltage), &(all_hk_data->EPS_hk.vBatt), sizeof(beacon_packet.battery_voltage));
+  memcpy(&(beacon_packet.battery_input_current), &(all_hk_data->EPS_hk.curBattIn), sizeof(beacon_packet.battery_input_current));
+  for (i = 0; i < 10; i++) {
+    memcpy(&(beacon_packet.current_channels[i]), &(all_hk_data->EPS_hk.curOutput[i]), sizeof(uint16_t));
+  }  
+  memcpy(&(beacon_packet.output_states), &(all_hk_data->EPS_hk.outputStatus), sizeof(uint16_t));
+  for (i = 0; i < 10; i++) {
+    memcpy(&(beacon_packet.output_faults[i]), &(all_hk_data->EPS_hk.outputFaultCnt[i]), sizeof(uint8_t));
+  }  
+  memcpy(&(beacon_packet.EPS_boot_count), &(all_hk_data->EPS_hk.bootCnt), sizeof(beacon_packet.EPS_boot_count));
+  
+  //Last EPS reset reason
+  typedef enum {
+    Power_on = 0, //After power supply removal or hard reset
+    IWDG = 1,     //Internal watchdog reseted system due to software process hanging or MCU hardware malfunction
+    NRST = 2,     //MCU reset pin driven low
+    Software = 3, //Software reset by command
+  } EPS_reset_TypeDef;
+
+  EPS_reset_TypeDef EPS_last_reset_reason;
+  if (all_hk_data->EPS_startup_hk.last_reset_reason_reg = 0x0C800000) {EPS_last_reset_reason = Power_on;}
+  if (all_hk_data->EPS_startup_hk.last_reset_reason_reg = 0x24000000) {EPS_last_reset_reason = IWDG;}
+  if (all_hk_data->EPS_startup_hk.last_reset_reason_reg = 0x24000000) {EPS_last_reset_reason = NRST;}
+  if (all_hk_data->EPS_startup_hk.last_reset_reason_reg = 0x14000000) {EPS_last_reset_reason = Software;}
+
+  //TEST: eps_last_reset_reason added, define the function based on EPS ICD manual
+  memcpy(&(beacon_packet.eps_last_reset_reason), &EPS_last_reset_reason, sizeof(uint8_t));
+  
+  /*-------Watchdog-------*/
+  //TODO: More work to be done on watchdogs
+  memcpy(&(beacon_packet.gs_wdt), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint8_t));
+  memcpy(&(beacon_packet.obc_wdt), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint16_t));
+  memcpy(&(beacon_packet.gs_wdt_expr), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint8_t));
+  memcpy(&(beacon_packet.obc_wdt_expr), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint16_t));
+  
+  /*-------Temperatures-------*/
+  //Onboard battery pack temp location defined in ICD
+  memcpy(&(beacon_packet.temps[0]), &(all_hk_data->EPS_hk.temp[8]), sizeof(int8_t));
+
+  //TODO: Athena MCU temperature should not be stored as long data type since data size can change depending on the computer

```

Comment:
Complete this todo. use int\<n\>_t to specify the length of the integer

Reply:
if you look at the next line, I think it needs to be fixed at the driver, and the driver will be changed later because we will be using a different chip. I've created a clickup task for this.

Reply:
thank you